app/forms/household_relationship_form.py

```diff
@@ -0,0 +1,91 @@
+from flask_wtf import FlaskForm
+from wtforms import FieldList, SelectField
+from wtforms.widgets import ListWidget, RadioInput
+
+from app.forms.fields import build_choices, get_mandatory_validator
+from app.helpers.schema_helper import SchemaHelper
+from app.jinja_filters import format_household_member_name
+
+from werkzeug.datastructures import MultiDict
+
+
+def build_relationship_choices(answer_store, group_instance):
+    """
+    A function to build a list of tuples of as yet undefined person relationships
+
+    :param answer_store: The answer store to use for current answers
+    :param group_instance: The instance of the group being iterated over
+    :return:
+    """
+    household_answers = answer_store.filter(block_id='household-composition')
+
+    first_names = []
+    last_names = []
+
+    for answer in household_answers:
+        if answer['answer_id'] == 'first-name':
+            first_names.append(answer['value'])
+        if answer['answer_id'] == 'last-name':
+            last_names.append(answer['value'])
+
+    household_members = []
+
+    for first_name, last_name in zip(first_names, last_names):
+        household_members.append({
+            'first-name': first_name,
+            'last-name': last_name,
+        })
+    remaining_people = household_members[group_instance + 1:] if group_instance < len(household_members) else []
+
+    current_person_name = format_household_member_name([
+        household_members[group_instance]['first-name'],
+        household_members[group_instance]['last-name'],
+    ])
+
+    choices = []
+
+    for remaining_person in remaining_people:
+        other_person_name = format_household_member_name([
+            remaining_person['first-name'],
+            remaining_person['last-name'],
+        ])
+        choices.append((current_person_name, other_person_name))
+
+    return choices
+
+
+def generate_relationship_form(block_json, number_of_entries, data, error_messages):
+    class HouseHoldRelationshipForm(FlaskForm):
+        def map_errors(self):
+            ordered_errors = []
+
+            if len(self.errors) > 0:
+                for answer_id, error_list in self.errors.items():
+                    for errors in error_list:
+                        for error in errors:
+                            ordered_errors.append((answer_id, error))
+
+            return ordered_errors
+
+        def answer_errors(self, input_id):
+            return [error[1] for error in self.map_errors() if input_id == error[0]]
+
+    answer = SchemaHelper.get_first_answer_for_block(block_json)
+
+    field = FieldList(SelectField(
+        label=answer.get('guidance'),
+        description=answer.get('label'),
```

Comment:
I commented on this but GitHub decided it was out of date for some reason; in the change to use `.get()` the default of `''` was removed, is that intentional? This will return None now if they aren't found.